ecl/hqlcpp/hqlckey.cpp

```diff
@@ -1407,7 +1409,7 @@ ABoundActivity * HqlCppTranslator::doBuildActivityKeyedJoinOrDenormalize(BuildCt
         flags.append("|JFindexoptional");
     if (info.needToExtractJoinFields())
         flags.append("|JFextractjoinfields");
-    if (expr->hasAttribute(unorderedAtom))
+    if (!isOrdered(expr))
```

Comment:
Yes.  hasOrderedAttribute would be wrong, since ordered(false) would be taken as ordered.